src/web/twig/Extension.php

```diff
@@ -944,11 +944,20 @@ public function shuffleFunction($arr)
      * @param bool|null $namespace Whether class names and IDs within the SVG
      * should be namespaced to avoid conflicts with other elements in the DOM.
      * By default the SVG will only be namespaced if an asset or markup is passed in.
-     * @param string|null $class A CSS class name that should be added to the `<svg>` element.
+     * @param array|string|null $attributes A list of attributes that should be added to the root `<svg>` element.
+     * @param array|null $elements Any elements that should be added under the  root`svg` element
```

Comment:
This should be just be `$title`, which can be `string|null`. No need to support additional nested elements AFAIK. (And move this along with `$attr` to be after `$class`.)

Reply:
What about `<desc>` tags? The underlying topic of #4660 and #3937 was accessibility so wouldn't it be prudent to cover both elements - not just `<title>`?

Reply:
Maybe both `$title` and `$desc` then?

Reply:
It's your call but personally, I'm not a fan. That means we end up with

- the duplication of running the same code twice or
- A loop using a hardcoded argument that we may as well make open to public use. 

It's a matter of preference so... Only thing I will say is if any accessibility tags are ever added in the future that will require another argument to the function.

Reply:
Also, the `<metadata>` tag could be supported by the `$elements` argument. Not sure what it's use is though?

Reply:
Ultimately I think you’re right that we shouldn’t limit the types of elements that can be added. But at that point it really doesn’t need to be a feature of the `svg()` function anymore. So I just [added](https://github.com/craftcms/cms/commit/5ee3d5f91fa90da2c36c8757224af767236efb1b) new generic `|append` and `|prepend` Twig filters, in the same vein as the new `|attr` filter, which can be used on SVGs or any other HTML element.

Full explanation here: https://github.com/craftcms/cms/issues/3937#issuecomment-522548611

Reply:
2 birds one stone - nice one!